src/umap.cpp

```diff
@@ -197,70 +169,13 @@ void umap_cfg_set_bufsize( int page_bufsize )
     umap_page_bufsize = page_bufsize;
 }
 
-_umap::_umap(  void* _mmap_addr, size_t _mmap_length, int _mmap_fd)
-    :   segment_address{_mmap_addr}, segment_length{_mmap_length},
-        backingfile_fd{_mmap_fd}, time_to_stop{false}, fault_count{0}, next_page_alloc_index{0}
-{
-    page_buffer_size = umap_page_bufsize;
-    if ((userfault_fd = syscall(__NR_userfaultfd, O_CLOEXEC | O_NONBLOCK)) < 0) {
-        perror("userfaultfd syscall not available in this kernel");
-        throw -1;
-    }
-
-    struct uffdio_api uffdio_api = { .api = UFFD_API, .features = 0 };
-
-    if (ioctl(userfault_fd, UFFDIO_API, &uffdio_api) == -1) {
-        perror("ioctl(UFFDIO_API)");
-        throw -1;
-    }
-
-    if (uffdio_api.api != UFFD_API) {
-        cerr << __FUNCTION__ << ": unsupported userfaultfd api\n";
-        throw -1;
-    }
-
-    struct uffdio_register uffdio_register = {
-        .range = { .start = (uint64_t)segment_address, .len = segment_length },
-        .mode = UFFDIO_REGISTER_MODE_MISSING | UFFDIO_REGISTER_MODE_WP
-    };
-
-    if (ioctl(userfault_fd, UFFDIO_REGISTER, &uffdio_register) == -1) {
-        perror("ioctl/uffdio_register");
-        close(userfault_fd);
-        throw -1;
-    }
-
-    enable_wp_on_pages_and_wake((uint64_t)segment_address, segment_length / page_size);
-
-    if ((uffdio_register.ioctls & UFFD_API_RANGE_IOCTLS) != UFFD_API_RANGE_IOCTLS) {
-        cerr << "unexpected userfaultfd ioctl set\n";
-        close(userfault_fd);
-        throw -1;
-    }
-
-    posix_memalign((void**)&tmppagebuf, (size_t)512, page_size);
-    if (tmppagebuf == nullptr) {
-        cerr << "Unable to allocate 512 bytes for temporary buffer\n";
-        close(userfault_fd);
-        throw -1;
-    }
-
-    umap_page ump;
-    pages_in_memory.resize(page_buffer_size, ump);
-
-    listener = new thread{&_umap::uffd_handler, this};      // Start our userfaultfd listener
-}
 //--------------------------for multi-file support----------------------
-_umap::_umap(void* _mmap_addr, size_t _mmap_length, int _mmap_fd,int* file_list, off_t data_offset, off_t frame)
+_umap::_umap(void* _mmap_addr, size_t _mmap_length, int num_backing_file,umap_backing_file* backing_files)
     :   segment_address{_mmap_addr}, segment_length{_mmap_length},
-        backingfile_fd{_mmap_fd}, time_to_stop{false}, fault_count{0}, next_page_alloc_index{0},
-        number_file{_mmap_fd}, fd_list{file_list}, fits_offset{data_offset}
+	      num_bk_files{num_backing_file},bk_files{backing_files},
```

Comment:
These need to be replaced with new allocations of backing_file objects into the vector mentioned above..   We don't want the library to depend upon pointers from the "C" callers since the library does not "own" them.  Better to allocate our own objects (and put them into a vector) and allow the caller to do what it wishes with the pointers it owns.